### PR TITLE
Added a wizard to create infrastructure provider

### LINKED ISSUE #13: Create UI to create Providers
Need to create UI to create and edit Platform providers

### PR DESCRIPTION
**Pull request by eva-vashkevich**
Fixes #13 
Added a wizard to create infrastructure provider.
Need to clarify with the BE what to do in the case of a custom provider and how to handle required variables ie for aws.

### CODE REVIEW

File: pkg/capi/edit/ProviderConfig.vue [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+        :cancel="cancelCredential"
+        :showing-form="showForm"
+        class="mb-40"
+      />
+      <h3 class="mb-20">
+        <t k="capi.provider.features.title" />
+      </h3>
+      <div class="mb-40">
+        <Checkbox
+          v-model="value.spec.features.clusterResourceSet"
```
Comment (mantis-toboggan-md) [inline pkg/capi/edit/ProviderConfig.vue:210]: The default providers installed w/ turtles don't have `spec.features` defined so these checkboxes throw an error when editing them. There are different ways of handling that; I think the fastest would be to set `features: {}` when initializing the component but the more correct way is probably to change the value prop on checkboxes to something that wont break if spec.features is undefined and update the  input event handler to set spec.features if not defined

File: pkg/capi/edit/turtles-capi.cattle.io.capiprovider.vue [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    selectType(type: string) {
+      this.subType = type;
+      this.$emit('set-subtype', this.$store.getters['i18n/withFallback'](`cluster.provider."${ type }"`, null, type));
+    },
+  },
+});
+</script>
+
+<template>
+  <CruResource
```
Comment (mantis-toboggan-md) [inline pkg/capi/edit/turtles-capi.cattle.io.capiprovider.vue:119]: It seems a little off to have CruResource both here and in ProviderConfig; what is it needed for in this outer component?

**Review by mantis-toboggan-md — CHANGES_REQUESTED** [review]

File: pkg/capi/assets/images/providers/aws.svg
[the remark this answers is not in the document]
Comment (eva-vashkevich) [reply pkg/capi/assets/images/providers/aws.svg:1]: I ran into an issue that I needed to provide my own image for docker but I couldn't make the format work. require didn't like dynamic string, so I've decided it's easier to have our own.

File: pkg/capi/edit/turtles-capi.cattle.io.capiprovider.vue (hunk 2 above)
Comment (eva-vashkevich) [reply pkg/capi/edit/turtles-capi.cattle.io.capiprovider.vue:119]: It allows to cancel while selecting provider

File: pkg/capi/edit/ProviderConfig.vue [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+  },
+  methods:  {
+    initSpecs() {
+      if ( !this.value.spec ) {
+        set(this.value, 'spec', clone(defaultSpec));
+      }
+      if ( this.provider !== 'custom') {
+        set(this.value.spec, 'name', this.provider); // Defines the provider kind to provision.
+      }
+      set(this.value.spec.configSecret, 'name', this.generateName(this.provider)); // Defines the name of the secret that will be created or adjusted based on the content of the spec.features and spec.variables.
```
[the remark this answers is not in the document]
Comment (eva-vashkevich) [reply pkg/capi/edit/ProviderConfig.vue:113]: we can show it to the user, but I don't think it makes sense to do

File: pkg/capi/edit/ProviderConfig.vue (hunk 1 above)
Comment (eva-vashkevich) [reply pkg/capi/edit/ProviderConfig.vue:210]: I think we will need to discuss how to handle non-standard providers in general tomorrow. I will just take out that options until it is decided (hopefully, tomorrow).

File: pkg/capi/edit/ProviderConfig.vue (hunk 3 above)
[the remark this answers is not in the document]
Comment (mantis-toboggan-md) [reply pkg/capi/edit/ProviderConfig.vue:113]: I agree that it doesn't make sense to expose it. It seems like something that should just be auto-generated on BE if the user doesn't set it, but we'll have to ask.

**Comment by eva-vashkevich** [comment]
Credentials mapping is not available by BE yet, so the missing variable error is currently expected.

File: pkg/capi/models/turtles-capi.cattle.io.capiprovider.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  _YAML,
+  AS,
+} from '@shell/config/query-params';
+
+export default class CapiCluster extends SteveModel {
+  get canEditYaml() {
+    return false;
+  }
+
+  get canUpdate() {
```
[the remark this answers is not in the document]
Comment (eva-vashkevich) [reply pkg/capi/models/turtles-capi.cattle.io.capiprovider.js:12]: Yes, but editing has been throwing an error and I've checked with Richard and we agreed to just disable it for now. Let me check if it works with 0.5.0

**Review by mantis-toboggan-md — APPROVED** [review]
LGTM

**Pull request merged**